webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/healthcheck/HealthcheckService.scala

```diff
@@ -25,6 +25,21 @@ object HealthcheckResult {
   final case class Critical(val msg: String) extends HealthcheckResult
 }
 
+object HealthcheckUtils {
+  def compareCheck(c1: HealthcheckResult, c2: HealthcheckResult): Boolean = {
+    (c1, c2) match {
+      case (_:Critical, _:Ok)       => true
+      case (_:Critical, _:Warning)  => true
```

Comment:
equiv to:
```
       case (a:Critical, b:Critical) => a.msg <= b.msg
       case (_:Critical, _)          => true
       case (_, _:Critical)          => false
       case (a:Warning, b:Warning)   => a.msg <= b.msg
etc
```

If you want it too be shorter, you can add an "index: Int" in `HealthcheckResult` and just do:
```
       case (a, b) if(a.index == b.index) => a.msg <= b.msg
       case (a, b)  => a.index <= b.index
```